drivers/hmis/app/models/hmis/form/definition.rb

```diff
@@ -325,6 +325,16 @@ def self.for_project(project:, role:, service_type: nil)
     where(status: PUBLISHED)
   end
 
+  validate :validate_external_form_object_key
+  def validate_external_form_object_key
+    return unless role == 'EXTERNAL_FORM' && external_form_object_key.present?
+
+    # Ensure that external_form_object_key is not reused across forms with different identifiers
+    object_key_taken = Hmis::Form::Definition.where(external_form_object_key: external_form_object_key).
+      where.not(identifier: identifier).exists?
+    errors.add(:external_form_object_key, 'has already been taken') if object_key_taken
+  end
```

Comment:
This is fine. Maybe could be expressed more concisely with [active record validations](https://guides.rubyonrails.org/active_record_validations.html#uniqueness) but I haven't tested it
```suggestion
  validates :external_form_object_key, uniqueness: { scope: :identifier},
    if: -> { role == 'EXTERNAL_FORM' && external_form_object_key.present? }
```

Reply:
I tried to come up with something using active record validations, but couldn't get there. Your example doesn't quite hit the mark, because there CAN be multiple `external_form_object_key`s with the same identifier, they just can't be reused _across_ identifiers. (At least, I think that makes sense. It could get very confusing if you have different forms that both publish to the same place).

Valid:
```
my-identifier,object_key_1,retired,v1
my-identifier,object_key_1,published,v2
my-identifier,object_key_1,draft,v3
```

Invalid:
```
identifier-1,object_key_1,published,v1
identifier-2,object_key_1,draft,v2
```